source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,316 @@
+            if (input1.eof())
+            break;
+            latitudes_iso.push_back(latitude_iso);
+            longitudes_iso.push_back(longitude_iso);
+            depths_iso.push_back(depth_iso);
+
+            /** Find first 2 numbers that are different to use in
+             * calculating half the difference between each position as delta.
+             */
+            if (count < 2 )
+              {
+                count ++;
+              }
+            if (count == 2)
+              {
+                if ((std::fabs(latitudes_iso[0] - latitudes_iso[1]) > 1e-9) && (std::fabs(longitudes_iso[0] - longitudes_iso[1]) > 1e-9))
+                  {
+                    /** Stop program if file formatted incorrectly.
+		             */
+                    std::cout << ""<< std::endl;
+                    throw std::ios_base::failure("Adiabatic boundary file not formatted correctly. " + adiabatic_boundary_file_name + "Make sure you have lat, lon, value with lat. or lon. varying.");
+                  }
+                if ((std::fabs(latitudes_iso[0] - latitudes_iso[1]) < 1e-9) && (std::fabs(longitudes_iso[0] - longitudes_iso[1]) < 1e-9))
+                  {
+                    /** Stop program if file formatted incorrectly.
+		             */
+                    std::cout << ""<< std::endl;
+                    throw std::ios_base::failure("Adiabatic boundary file not formatted correctly. " + adiabatic_boundary_file_name + "Make sure you have lat, lon, value with lat. or lon. varying.");
+                  }
+
+                if (std::fabs(latitudes_iso[0] - latitudes_iso[1]) > 1e-9)
+                  {
+                    /** Calculate delta as half the distance between points.
+		             */
+                    delta = std::fabs((0.5)*(latitudes_iso[0] - latitudes_iso[1]));
+                    /** If flag is 0 then longitudes grouped and we calculate delta from latitudes
+		             */
+                    data_flag = 0;
+                  }
+                else
+                  {
+                    /** Calculate delta as half the distance between points.
+		             */
+                    delta = std::fabs((0.5)*(longitudes_iso[0] - longitudes_iso[1]));
+                    /** If flag is 1 then latitudes are grouped and we calculate delta from longitudes
+                     */
+		    data_flag = 1;
+                  }
+                std::cout << ""<< std::endl;
+                std::cout<<"Adiabatic boundary data interaval  delta = "<< delta << std::endl;
+                std::cout << ""<< std::endl;
+                std::cout<<"Resolution of input data in meters is approximately = "<< delta*111*2 << std::endl;
+                std::cout << ""<< std::endl;
+                count++;
+              }
+          }
+
+    /** Calculate the number of unique longitudes or latitudes from the adiabatic boundary file.
+     */
+	double c,d;
+        int count3;
+        count3 = 0;
+        c = 0;
+        d = 0;
```

Comment:
This style is hard to read. What do variables `c,d,count3` actually stand for? Could you rename them to something that makes it easier to understand what they refer to?